platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/events/BaseEventHashedData.java

```diff
@@ -222,7 +226,14 @@ public void deserialize(
         } else {
             softwareVersion = SoftwareVersion.NO_VERSION;
         }
-        creatorId = in.readLong();
+        // FUTURE WORK: The creatorId should be a selfSerializable NodeId at some point.
+        // Changing the event format may require a HIP.  The old format is preserved for now.
+        final long serializedCreatorId = in.readLong();
```

Comment:
Maybe add a static deserialize method to `NodeId` with this logic

Reply:
Added